lib/utils/types.ts

```diff
@@ -26,6 +32,23 @@ export const createEnumType = <E>(e: {}, name: string): t.Type<any, E> => {
   );
 };
 
+/**
+ * Creates an io-ts Type from a ReadonlySet
+ */
+export const readonlySetType = <S, E>(
+  o: t.Type<S, E>,
+  name: string
+): t.Type<any, ReadonlySet<E>> => {
+  const arrayType = t.readonlyArray(o, name);
+  return new t.Type<any, ReadonlySet<E>>(
+    name,
+    (v): v is ReadonlySet<E> => v instanceof Set && arrayType.is(Array.from(v)),
+    (s, c) =>
+      arrayType.is(s) ? t.success(new Set(Array.from(s))) : t.failure(s, c),
```

Comment:
ensure to 
`import { Set } from "json-set-map";` at the beginning of the file
otherwise you get unserializable Set()